src/herbpy/herbrobot.py

```diff
@@ -242,35 +319,187 @@ def CloneBindings(self, parent):
+
+        for controller in active_controllers:
+            controller.SetPath(traj)
+
+        if defer is True:
+            import time
+            import trollius
+
+            @trollius.coroutine
+            def do_poll():
+                time_stop = time.time() + (timeout if timeout else numpy.inf)
+
+                while time.time() <= time_stop:
+                    is_done = all(controller.IsDone()
+                                  for controller in active_controllers)
+                    if is_done:
+                        raise trollius.Return(traj)
 
-        active_manipulators = self.GetTrajectoryManipulators(traj)
+                    yield trollius.From(trollius.sleep(period))
 
-        for manipulator in active_manipulators:
-            manipulator.ClearTrajectoryStatus()
+                raise trollius.Return(None)
 
-        value = super(HERBRobot, self).ExecuteTrajectory(traj, *args, **kwargs)
+            return trollius.async(do_poll())
+        elif defer is False:
+            prpy.util.WaitForControllers(active_controllers, timeout=timeout)
+            return traj
+        else:
+            raise ValueError('Received unexpected value "{:s}" for defer.'
+                             .format(str(defer)))
+
+    def ExecuteTrajectory(self, traj, *args, **kwargs):
+        # from prpy.exceptions import TrajectoryAborted
 
-        for manipulator in active_manipulators:
-            status = manipulator.GetTrajectoryStatus()
-            if status == 'aborted':
-                raise TrajectoryAborted()
+        value = self._ExecuteTrajectory(traj, *args, **kwargs)
+
+        # TODO meaningful to do this check here?
+        # if so can be done on value future
+        #
+        # for manipulator in active_manipulators:
+        #     status = manipulator.GetTrajectoryStatus()
+        #     if status == 'aborted':
+        #         raise TrajectoryAborted()
 
         return value
 
     # Inherit docstring from the parent class.
     ExecuteTrajectory.__doc__ = Robot.ExecuteTrajectory.__doc__
 
-    def SetStiffness(self, stiffness):
+    def SetStiffness(self, stiffness, manip=None):
         """Set the stiffness of HERB's arms and head.
-        Zero is gravity compensation, one is position control. Stifness values
-        between zero and one are experimental.
-        @param stiffness value between zero and one
+        Stiffness False/0 is gravity compensation and stiffness True/(>0) is position
+        control.
+        @param stiffness boolean or decimal value 0.0 to 1.0
         """
-        self.head.SetStiffness(stiffness)
-        self.left_arm.SetStiffness(stiffness)
-        self.right_arm.SetStiffness(stiffness)
+        if isinstance(stiffness, numbers.Number) and not (0 <= stiffness <= 1):
+            raise Exception('Stiffness must be boolean or decimal in the range [0, 1];'
+                            'got {}.'.format(stiffness))
```

Comment:
"Decimal" is confusing here - floating point numbers are not technically decimals. :smile: